storage/src/test/java/org/apache/kafka/storage/internals/log/RemoteIndexCacheTest.java

```diff
@@ -0,0 +1,1207 @@
+        long entrySizeInBytes = entry.entrySizeBytes();
+        Utils.closeQuietly(cache, "RemoteIndexCache created for estimating entry size");
+        return entrySizeInBytes;
+    }
+
+    private void mockRsmFetchIndex(RemoteStorageManager rsm) throws RemoteStorageException {
+        when(rsm.fetchIndex(any(RemoteLogSegmentMetadata.class), any(IndexType.class))).thenAnswer(ans -> {
+            RemoteLogSegmentMetadata metadata = ans.getArgument(0);
+            IndexType indexType = ans.getArgument(1);
+            OffsetIndex offsetIdx = createOffsetIndexForSegmentMetadata(metadata, tpDir);
+            TimeIndex timeIdx = createTimeIndexForSegmentMetadata(metadata, tpDir);
+            TransactionIndex txnIdx = createTxIndexForSegmentMetadata(metadata, tpDir);
+            maybeAppendIndexEntries(offsetIdx, timeIdx);
+            return switch (indexType) {
+                case OFFSET -> new FileInputStream(offsetIdx.file());
+                case TIMESTAMP -> new FileInputStream(timeIdx.file());
+                case TRANSACTION -> new FileInputStream(txnIdx.file());
+                case LEADER_EPOCH -> null; // leader-epoch-cache is not accessed.
+                case PRODUCER_SNAPSHOT -> null; // producer-snapshot is not accessed.
+            };
+        });
+    }
+
+    private void createCorruptOffsetIndexFile(File dir) throws FileNotFoundException {
+        PrintWriter pw = new PrintWriter(remoteOffsetIndexFile(dir, rlsMetadata));
+        pw.write("Hello, world");
+        // The size of the string written in the file is 12 bytes,
+        // but it should be multiple of Offset Index EntrySIZE which is equal to 8.
+        pw.close();
+    }
+
+    private void createCorruptTimeIndexOffsetFile(File dir) throws FileNotFoundException {
+        PrintWriter pw = new PrintWriter(remoteTimeIndexFile(dir, rlsMetadata));
+        pw.write("Hello, world1");
+        // The size of the string written in the file is 13 bytes,
+        // but it should be multiple of Time Index EntrySIZE which is equal to 12.
+        pw.close();
+    }
+
+    private void createCorruptedIndexFile(RemoteStorageManager.IndexType indexType, File dir) throws IOException {
+        if (indexType == RemoteStorageManager.IndexType.OFFSET) {
+            createCorruptOffsetIndexFile(dir);
+        } else if (indexType == IndexType.TIMESTAMP) {
+            createCorruptTimeIndexOffsetFile(dir);
+        } else if (indexType == IndexType.TRANSACTION) {
+            createCorruptTxnIndexForSegmentMetadata(dir, rlsMetadata);
+        }
+    }
+
+    private Optional<Path> getIndexFileFromRemoteCacheDir(RemoteIndexCache cache, String suffix) {
+        try {
+            return Files.walk(cache.cacheDir().toPath())
+                    .filter(Files::isRegularFile)
+                    .filter(path -> path.getFileName().toString().endsWith(suffix))
+            .findAny();
+        } catch (UncheckedIOException | IOException exc) {
```

Comment:
Are those exceptions expected? If not, we should use `assertDoesNotThrow` instead of swallowing them

Reply:
`Files.walk()` throws `IOException` but it seems we can remove `UncheckedIOException`